src/cgeohash.h

```diff
@@ -18,7 +18,7 @@ struct DecodedHash {
 };
 
 // Encode a pair of latitude and longitude into geohash
-std::string encode(double latitude, double longitude, int precision);
+std::string encode(double latitude, double longitude, unsigned int precision);
```

Comment:
gah! my appologies. it seems that my for fell out of sync with your repo and therefore there are more changes here than just mine.  ugh.  i can try and clean this up later this week.